### PR TITLE
Fsa22 v2 222 component components grid

### PR DESCRIPTION
**Pull request by tisidro**
### Description:
Components Grid

### Spec:
Designs: [Designs](DESIGN_URL)

See Story: [FSA22V2-222](https://sparkbox.atlassian.net/browse/FSA22V2-222?atlOrigin=eyJpIjoiZWY2YjQ5OGZlZTA5NDI4ZmIxZGYzZTFmNGYwYTIxOTkiLCJwIjoiaiJ9)

### Validation:
Create grid layout for component cards, render component cards in layout.

* [ ] This PR has code changes, and our linters still pass.
* [ ] This PR affects production code, so it was browser tested (see below).
* [ ] This PR has new code, so new tests were added or updated, and they pass.

#### To Validate:

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
2. Pull down all related branches.
3. In terminal: `npm run dev`.
4. Check [localhost:3000](http://localhost:3000/)for changes.
5. Run `npm run lint`.
6. Run axe Devtools on affected pages.
7. Confirm all tests pass.

- Please check that grid layout is responsive and follows parameters on Jira ticket.
- As soon as Marissa has her components card approved, I will drop the components in place and ask for re-review but I figured it would be good to get this one into review. :)
---

#### Browser Testing

In these browsers, behavior & design closely match original specifications. A user is able to access all content and functionality, including the usability of required assistive devices, such as keyboard and screenreader.

**macOS**

* [ ] Safari (last 2 major versions)
* [ ] Chrome (last 6 months)
* [ ] Firefox (last 6 months)

**Windows**

* [ ] Chrome (last 6 months)
* [ ] Firefox (last 6 months)
* [ ] Edge (last 6 months)

**Mobile**

* [ ] Safari (last 2 major versions)
* [ ] Chrome on Android (last 6 months)
* [ ] Firefox on Android (last 6 months)

### CODE REVIEW

File: package.json
Comment (marissahuysentruyt) [inline package.json:29]: Can I ask why you deleted this? If we don't need it, sweet, but I found I still needed it to get my tests to pass. I have been doing something funky though, so I'd just love to know why we don't need anymore!

File: src/pages/index.jsx
Comment (marissahuysentruyt) [inline src/pages/index.jsx:10]: This is where Jared's `Layout` should go.

Comment (tisidro) [reply src/pages/index.jsx:10]: Agreed, we just haven't added that in yet but it's coming in the next commit! :)

File: package.json
Comment (tisidro) [reply package.json:29]: Weird that it was deleted, that was not on purpose. For sure will add it back in!

Comment (tisidro) [reply package.json:29]: If you please go ahead and add that as a required in your review I'll do a fix up, then you can approve (I'll ping you via DM once it's updated)

File: src/styles/components/_component-grid.scss
Comment (marissahuysentruyt) [inline src/styles/components/_component-grid.scss:11]: I know you and I talk about this, but I think it just got missed. There's something responsive issues with grid again, so I just wanted to pass along the fixes we talked about: 
```suggestion
..components {
  display: grid;
  grid-template-columns: 1fr;
  gap: 1rem;

  &-grid {
    width: 100%;
    padding: 0;
  }

  &-grid__item {
    width: 100%;
  }

  &-grid {
    &__item {
      display: flex;
      justify-content: center;

      &--accordion {
        grid-area: accordion;
      }

      &--dialog {
        grid-area: dialog;

```

File: src/styles/components/_componentCard.scss
Comment (marissahuysentruyt) [inline src/styles/components/_componentCard.scss:13]: What if under `.component-card` you also added `width: 100%`? It would be on line 8.

File: src/styles/components/_home.scss
Comment (marissahuysentruyt) [inline src/styles/components/_home.scss:12]: Took us long enough, but we found these!!! They made all the difference!

File: src/styles/globals.scss
Comment (marissahuysentruyt) [inline src/styles/globals.scss:12]: I think if you run Bennett's `npm run lint` you're going to get flagged for all of the single quotes, and not having the font names on single lines.

**Review by marissahuysentruyt — CHANGES_REQUESTED** [review]

**Comment by marissahuysentruyt** [comment]
This is looking so good!! I'm so happy we figured out the responsive issues!

File: src/styles/components/_component-grid.scss
Comment (tisidro) [reply src/styles/components/_component-grid.scss:11]: Perfect! I added this and it helped center the layout :)

File: src/styles/components/_componentCard.scss
Comment (tisidro) [reply src/styles/components/_componentCard.scss:13]: I didn't notice a change, but it didn't hurt either!

File: src/styles/globals.scss
Comment (tisidro) [reply src/styles/globals.scss:12]: I'll send that up in the next commit though.

**Comment by empee3** [comment]
Hi @tisidro! I'm not sure if you're all focusing on the mobile version yet, but I did notice that the grid isn't centered on smaller screens. This is because there is some default left padding on the `components-grid` `ul`.  If you add some padding to the right as well, this should center everything nicely.  If mobile isn't a concern yet, just keep this in mind for later.

 
Screen Shot 2022-10-14 at 12 33 41 PM

File: src/pages/index.jsx
Comment (empee3) [inline src/pages/index.jsx:34]: This is excellent `alt` text!

File: src/styles/globals.scss
Comment (empee3) [reply src/styles/globals.scss:12]: Yep, I think all of the build failures are just due to linting errors.

Screen Shot 2022-10-14 at 12 26 22 PM

File: src/pages/index.jsx
Comment (tisidro) [reply src/pages/index.jsx:34]: Thanks! As far as your advice for mobile--yes it needs to be mobile responsive so I will fix that padding as you suggested. thank you!

File: src/styles/globals.scss
Comment (tisidro) [reply src/styles/globals.scss:6]: They should, I will change mine to camelCase :)

File: src/pages/index.jsx
Comment (dustin-jw) [reply src/pages/index.jsx:34]: Formatting note: I would expect each `ComponentCard` line to be on a new line inside of the `li` tags.

```jsx
<li>
  <ComponentCard />
</li>
```

File: src/styles/globals.scss
Comment (dustin-jw) [reply src/styles/globals.scss:6]: I'd go the other way and change `componentCard` to `component-card`. kebab-case is more consistent with how we use BEM.

File: src/pages/index.jsx
Comment (dustin-jw) [reply src/pages/index.jsx:34]: For the padding, I would remove the padding from the `ul`, rather than add `padding-right` to balance it out.

File: package.json
Comment (dustin-jw) [reply package.json:29]: I would not expect any `package.json` or `package-lock.json` changes to exist for this PR, so we need to get them matched up with `main` so they don't appear in the "Files changed list".

**Review by dustin-jw — CHANGES_REQUESTED** [review]

File: src/styles/components/_component-grid.scss
Comment (marissahuysentruyt) [inline src/styles/components/_component-grid.scss:19]: Could we also add `align-items: center` to the `.components-grid__item` (about line 20) that way the images don't elongate when the screen is an "in-between" size? The accordion and the dialog box in the screenshot below are a little funky. 
Screen Shot 2022-10-17 at 9 10 24 AM

File: package.json
Comment (marissahuysentruyt) [reply package.json:32]: This isn't the package.json still, so when I run npm run test I get a validation error. I think npm install --save-dev jest-environment-jsdom should do it.

File: src/styles/globals.scss
Comment (marissahuysentruyt) [inline src/styles/globals.scss:7]: Hopefully, once we get Prettier figured out, these single quotes will stay double for you!!

File: src/styles/globals.scss
Comment (marissahuysentruyt) [inline src/styles/globals.scss:15]: Same thing here: we'll need the double quotes to pass the linting tests!

**Review by marissahuysentruyt — CHANGES_REQUESTED** [review]

File: src/styles/components/_component-grid.scss
Comment (empee3) [inline src/styles/components/_component-grid.scss:8]: This might eventually be handled within a wrapper `<div>`, but if you add 
`margin: 0 auto;` here, it will center your component grid on the page so that the outside left border isn't against the left side of the screen. Otherwise, this looks great!

Comment (tisidro) [reply src/styles/components/_component-grid.scss:8]: Will do, thanks!

**Review by marissahuysentruyt — COMMENTED** [review]
Hey this is looking awesome! I think the only thing that is missing is the `lint:html` script that we noticed went missing yesterday.  I'm not sure how we handle getting that back in, whether it's in this pull request or a separate one.

**Comment by marissahuysentruyt** [comment]
Following up here on our Slack conversation about the missing `lint:html`. I only noticed it because I reviewed the PR that installed it! That PR was about Pa11y (commit f6ab7b40...). 

When I run `npm run lint:html` I get a bunch of errors. 
Screen Shot 2022-10-19 at 8 20 15 AM

In the package.json, I think we need a `lint:html` script that does a bunch of pa11y things (you might have to ask Bennett how to install those) and then additional commands get added to the `lint` script (so that it runs all of these linting scripts at once I think): 
Screen Shot 2022-10-19 at 8 20 44 AM

**Comment by tisidro** [comment]
Hi there @marissahuysentruyt , I just checked the package.json main branch and I do not see a lint:html script. I'm going to check in w/ Bennett to have him weigh in and see what I can do to add the script--but it needs to be on main (which it will be if I can get it in here and rebase/merge!) :)

image

**Comment by marissahuysentruyt** [comment]
> I just checked the package.json main branch and I do not see a lint:html script. 

I think it got deleted on accident with the last round of merges. Again, I only knew to look for it because I reviewed the PR that added that. It was on main at one point.

**Comment by tisidro** [comment]
> > I just checked the package.json main branch and I do not see a lint:html script.
> 
> I think it got deleted on accident with the last round of merges. Again, I only knew to look for it because I reviewed the PR that added that. It was on main at one point.

@marissahuysentruyt thank you--I just updated, tested, and committed it! I appreciate you pointing it out. Once this gets rebased/merged we will be good on main branch. :)

**Comment by marissahuysentruyt** [comment]
Terri, it's so close! I'm pretty sure you mentioned it, but those single quotes (I think they're in `global.scss`) are literally the only thing that I think would prevent this from getting into main right now!!!

**Review by dustin-jw — DISMISSED** [review]

**Review by marissahuysentruyt — DISMISSED** [review]
Get it!!!! 
Tests pass ✅
Linters pass ✅
Responsive ✅

Way to keep with this one!!! This was a giant challenge!!!

**Comment by tisidro** [comment]
I am planning the "component-components-grid Jira card merge to main" party as soon as I merge this! 🤣

File: package.json
Comment (dustin-jw) [inline package.json:11]: This is duplicated now, see how the same code is on line 14?

File: src/styles/components/index.scss
Comment (dustin-jw) [inline src/styles/components/index.scss:2]: This file snuck back in, get rid of it!

**Review by dustin-jw — CHANGES_REQUESTED** [review]
Looks like some unwanted changes came back from rebasing. You can add a step to your checklist to make sure that the staged files look like what we want the end result to be before running `git rebase --continue`.

File: src/styles/components/index.scss
Comment (tisidro) [reply src/styles/components/index.scss]: OK will do!

**Review by dustin-jw — DISMISSED** [review]

**Review by dustin-jw — APPROVED** [review]

**Pull request merged**
